components/switch/switch-mixin.js

```diff
@@ -0,0 +1,181 @@
+import '../colors/colors.js';
+import { css, html } from 'lit-element/lit-element.js';
+import { getUniqueId } from '../../helpers/uniqueId.js';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+
+export const SwitchMixin = superclass => class extends RtlMixin(superclass) {
+
+	static get properties() {
+		return {
+			disabled: { type: Boolean, reflect: true },
+			label: { type: String, reflect: true },
+			labelHidden: { type: Boolean, attribute: 'label-hidden', reflect: true },
+			on: { type: Boolean, reflect: true }
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				display: inline-block;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+
+			.d2l-switch-container {
+				background-color: #ffffff;
+				border: 2px solid transparent;
+				border-radius: 1rem;
+				box-sizing: border-box;
+				cursor: pointer;
+				display: inline-block;
+				font-size: 0;
+				line-height: 0;
+				outline-style: none;
+				padding: 0.1rem;
+				vertical-align: middle;
+			}
+			.d2l-switch-container:focus,
+			.d2l-switch-container:hover {
```

Comment:
You'll likely need a `focus()` method on the host that transfers focus inside to this element.

Reply:
Thanks, indeed... added this.